test/test_pipeline/test_tabular_regression.py

```diff
@@ -0,0 +1,268 @@
+class TestTabularRegression:
+    def _assert_pipeline_search_space(self, pipeline, search_space_updates):
+        config_space = pipeline.get_hyperparameter_search_space()
+        for update in search_space_updates.updates:
+            try:
+                assert update.node_name + ':' + update.hyperparameter in config_space
+                hyperparameter = config_space.get_hyperparameter(update.node_name + ':' + update.hyperparameter)
+            except AssertionError:
+                assert any(update.node_name + ':' + update.hyperparameter in name
+                           for name in config_space.get_hyperparameter_names()), \
+                    "Can't find hyperparameter: {}".format(update.hyperparameter)
+                hyperparameter = config_space.get_hyperparameter(update.node_name + ':' + update.hyperparameter + '_1')
+            assert update.default_value == hyperparameter.default_value
+            if isinstance(hyperparameter, (UniformIntegerHyperparameter, UniformFloatHyperparameter)):
+                assert update.value_range[0] == hyperparameter.lower
+                assert update.value_range[1] == hyperparameter.upper
+                if hasattr(update, 'log'):
+                    assert update.log == hyperparameter.log
+            elif isinstance(hyperparameter, CategoricalHyperparameter):
+                assert update.value_range == hyperparameter.choices
+
+    def test_pipeline_fit(self, fit_dictionary_tabular):
+        """This test makes sure that the pipeline is able to fit
+        given random combinations of hyperparameters across the pipeline"""
+
+        pipeline = TabularRegressionPipeline(
+            dataset_properties=fit_dictionary_tabular['dataset_properties'])
+        cs = pipeline.get_hyperparameter_search_space()
+
+        config = cs.sample_configuration()
+        pipeline.set_hyperparameters(config)
+        pipeline.fit(fit_dictionary_tabular)
+
+        # To make sure we fitted the model, there should be a
+        # run summary object with r2
+        run_summary = pipeline.named_steps['trainer'].run_summary
+        assert run_summary is not None
+
+        # Make sure that performance was properly captured
+        assert run_summary.performance_tracker['train_loss'][1] > 0
+        assert run_summary.total_parameter_count > 0
+        assert 'r2' in run_summary.performance_tracker['train_metrics'][1]
+
+        # Make sure a network was fit
+        assert isinstance(pipeline.named_steps['network'].get_network(), torch.nn.Module)
+
+    def test_pipeline_predict(self, fit_dictionary_tabular):
+        """This test makes sure that the pipeline is able to fit
+        given random combinations of hyperparameters across the pipeline"""
+        pipeline = TabularRegressionPipeline(
+            dataset_properties=fit_dictionary_tabular['dataset_properties'])
+
+        cs = pipeline.get_hyperparameter_search_space()
+        config = cs.sample_configuration()
```

Comment:
To check that the score is at least 0.8, change this to default configuration

Reply:
You mean that I should not just randomly sample and set a configuration?

Reply:
If you can that will be great :D, but for instance, this means that you won't have a consistent test to make sure regression code is working. With the default pipeline and after the randomness is removed, you will have a consistent check.

But maybe you get a very small learning rate, and then never achieve a good performance in the given budget.

Reply:
Makes sense.